Packager/CMakeLists.txt

```diff
@@ -20,48 +20,207 @@ set(MODULE_NAME WPEFramework${PLUGIN_NAME})
+    find_package(CompileSettingsDebug CONFIG REQUIRED)
+
+    # enabling the secure extension requires a license: https://www.hwaci.com/cgi-bin/see-step1
+    if (USE_SQLITE_SEE)
+        find_package(SqliteSee REQUIRED)
+    else()
+        find_package(Sqlite REQUIRED)
+    endif()
+
+    add_library(${MODULE_NAME} SHARED
+        Module.cpp
+        Packager.cpp
+        PackagerImplementation.cpp
+        PackagerExUtils.cpp
+        PackagerExImplementation.cpp)
+
+    if (CURL_FOUND)
+        include_directories(${CURL_INCLUDE_DIRS})
+        target_link_libraries(${MODULE_NAME} PRIVATE ${CURL_LIBRARIES})
+    else (CURL_FOUND)
+        message ("Curl/libcurl required.")
+    endif (CURL_FOUND)
+
+    add_definitions(${SQLITE_CFLAGS_OTHER} ${PLABELS_FLAGS})
+    link_directories(${SQLITE_LIBRARY_DIRS})
+
+
+    target_include_directories(${MODULE_NAME} PRIVATE ../helpers
+            ${SQLITE_INCLUDE_DIRS}
+            ${PLABELS_INCLUDE_DIRS}
+            ${GLIB_INCLUDE_DIRS})
+
+    if (libprovision_FOUND)
+        target_link_libraries(${MODULE_NAME}
+            PRIVATE
+                CompileSettingsDebug::CompileSettingsDebug
+                ${NAMESPACE}Plugins::${NAMESPACE}Plugins
+                LibOPKG::LibOPKG
+                ${LIBUUID_LIBRARY}
+                ${SQLITE_LIBRARIES}
+                ${GLIB_LIBRARIES}
+                libprovision::libprovision
+
+                )
+    else (libprovision_FOUND)
+        target_include_directories(${MODULE_NAME}
+            PRIVATE
+                ${LIBOPKG_INCLUDE_DIRS}
+                )
+
+        target_link_libraries(${MODULE_NAME}
+            PRIVATE
+                CompileSettingsDebug::CompileSettingsDebug
+                ${NAMESPACE}Plugins::${NAMESPACE}Plugins
+                ${LIBOPKG_LIBRARIES}
+                ${LIBUUID_LIBRARY}
+                ${SQLITE_LIBRARIES}
+                ${GLIB_LIBRARIES}
+                )
+    endif (libprovision_FOUND)
+
+    string(TOLOWER ${NAMESPACE} STORAGENAME)
+    install(TARGETS ${MODULE_NAME} 
+        DESTINATION lib/${STORAGENAME}/plugins)
+
+    configure_file(
+            "${CMAKE_CURRENT_SOURCE_DIR}/opkg.conf.in"
+            "${CMAKE_CURRENT_BINARY_DIR}/opkg.conf"
+            @ONLY)
+
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/opkg.conf
+            DESTINATION "/usr/share/${NAMESPACE}/${PLUGIN_NAME}")
+
+    write_config(${PLUGIN_NAME})
+
+######################################################################################################
+######################################################################################################
+##
+## Support for legacy PACKAGER
+##
+
+else (BUILD_WITH_PACKAGER_EX)
+
+    message("Building with support for legacy PACKAGER")
+
+    set(PLUGIN_NAME Packager)
+    set(MODULE_NAME ${NAMESPACE}${PLUGIN_NAME})
+
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+    find_path(LIBUUID_INCLUDE_DIR uuid.h PATH_SUFFIXES uuid)
```

Comment:
If I'm right uuid is not needed for legacy packager support?

Reply:
Correct - Thanks !!